### PR TITLE
[feat] add fed bn support

### PR DESCRIPTION
**Pull request by zhangxingmeng**
- refer to FedBN paper for handling the Batch Normalization layer in federated settings
- support tensorflow backend
- support torch backend
- add unittest for fedbn

### CODE REVIEW

File: secretflow/ml/nn/utils.py [hunk 1]
```diff
@@ -31,6 +31,25 @@ class BaseModule(ABC, nn.Module):
     def forward(self, x):
         pass
 
+    def get_weights_not_bn(self, return_numpy=False):
+        if not return_numpy:
+            return {k: v.cpu() for k, v in self.state_dict().items() if 'bn' not in k}
+        else:
+            weights_list = []
+            for k, v in self.state_dict().items():
+                if 'bn' not in k:
```
Comment (ian-huu) [inline secretflow/ml/nn/utils.py:40]: bn 这个 key 是个特殊的 key 吗，还是应该根据 layer 是不是 _BatchNorm 来判断

Comment (zhangxingmeng) [reply secretflow/ml/nn/utils.py:40]: 是一个字符串，比如`bn.weights`,`bn.bias`,`bn.running_var`

File: secretflow/ml/nn/fl/backend/tensorflow/fl_base.py [hunk 2]
```diff
@@ -217,12 +219,51 @@ def build_dataset_from_builder(
...
+        for layer in self.model.layers:
+            if not isinstance(layer, tf.keras.layers.BatchNormalization):
+                # This layer is not a BN layer, so update its weights from the provided list.
+                num_params = len(layer.trainable_weights)
+                new_weights = weights[index : index + num_params]
+                layer.set_weights(new_weights)
+                index += num_params  # Increment the index by the number of parameters in the current layer.
+            else:
+                # For BN layers, retain the current weights
+                updated_weights.extend(layer.get_weights())
```
[the remark this answers is not in the document]
Comment (zhangxingmeng) [reply secretflow/ml/nn/fl/backend/tensorflow/fl_base.py]: 没有用到，直接在层里面set了，这个删掉了

**Review by zhaocaibei123 — APPROVED** [review]

**Review by ian-huu — APPROVED** [review]
LGTM
